UT-Applicataion-Development-Association/Email-Sender-Server
# Adapt change from Eve's Branch and add basic Unittests on mail route

## Related Issue
[请将对应的Issue链接粘贴至此，若无则留空]
#3 #4 

## Type of Change


## Major Changes
[请简要列出该PR涵盖的主要改动]
Adapting code based on Eve that has been submitted and add unittest on the mail route.  

## Steps for Testing
[请简要说明测试该PR所需要的步骤]
A sample json is 
```
{"to":["liaolove@me.com"],
"subject": "Test Email", 
"recipients": ["liaolove@me.com"], 
"cc": [], 
"bcc": [], 
"body": "Test email", 
"attachment":[] }
```
On the postman, select request to POST and destination to http://localhost:8081/mail using the sample json provided above.  

The unittest for the mail route have been provided, neverthless, those tests are designed to verify behaviours including verifying user credential and smtp port auto correction which have not been implemented in the current version.  For now, we have to manully test the server using the postman.  

## Additional Context
[还有其他注意事项吗？]
I think the json we have right now missing the field of sender, properly we should add this field.  Currently, the config of smtp server and ports are hard coded using gmail.  For the future improvement, we should apply regex to extract sender's email thus we can automatically update server's config.  Those minor feature should be implmented before deploying next release version.
